File: [C]hashmap/[C]hashmap/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_idx = {}
        for i, num in enumerate(nums):
            if target - num in num_idx:
                return [i, num_idx[target - num]]
            num_idx[num] = i
        return []
        
# time O(n), due to traverse
# space O(n), due to hashmap
# using hashmap and store val

File: [C]hashmap/[C]hashmap/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> valToIdx = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (valToIdx.containsKey(target - nums[i])) {
                return new int[]{valToIdx.get(target - nums[i]), i};
            }
            valToIdx.put(nums[i], i);
        }
        return null;
    }
}

// time O(n), due to traverse
// space O(n), due to hashmap
// using hashmap and store val